src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1729,6 +1729,81 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
                 return;
             }
 
+            var usage = ticket.GetUsage();
+            var identifier = ticket.GetProperty(OpenIdConnectConstants.Extra.ClientId);
+            var audiences = ticket.GetAudiences();
+            if (string.Equals(usage, OpenIdConnectConstants.Usages.RefreshToken, StringComparison.Ordinal)) {
+                // For Refresh tokens
+                if ((clientNotification.IsSkipped &&
```

Comment:
I had this separate initially for readability, but I really didn't like the duplication of code (nit: I still don't for the repeated use of `await SendPayloadAsync(new JObject { [OpenIdConnectConstants.Claims.Active] = false });` but it's small enough that I can deal with it).

I don't see how the logging message isn't appropriate since the first condition is essentially `clientNotification.IsSkipped && ticket.ContainsProperty(OpenIdConnectConstants.Extra.ClientAuthenticated)`, meaning that the client is not authenticated AND the token was issued to an authenticated client - which means the token is for a different client and (probably) not the client that's trying to validate it. I guess coming from a different perspective, if that client is the one that's associated with the token but they're not authenticated, they should authenticate, but we don't want to provide that information to the caller since it says the token is valid and exists but that they just don't have access to it.